pkg/hs/urbit-king/lib/Urbit/Vere/Eyre/Wai.hs

```diff
@@ -199,36 +209,99 @@ sendResponse cb waitAct = do
 
 liveReq :: Ship -> TVar LiveReqs -> RAcquire e (Word64, STM RespAct)
 liveReq who vLiv = mkRAcquire ins del
- where
-  ins = atomically (newLiveReq who vLiv)
-  del = atomically . rmLiveReq vLiv . fst
-
-app
-  :: HasLogFunc e
-  => e
-  -> Ship
-  -> TVar LiveReqs
-  -> (Word64 -> ReqInfo -> STM ())
-  -> (Word64 -> STM ())
-  -> KingSubsite
-  -> W.Application
-app env who liv inform cancel sub req respond =
+  where
+    ins = atomically (newLiveReq who vLiv)
+    del = atomically . rmLiveReq vLiv . fst
+
+app ::
+  HasLogFunc e =>
+  e ->
+  Ship ->
+  TVar LiveReqs ->
+  (Word64 -> ReqInfo -> STM ()) ->
+  (Word64 -> STM ()) ->
+  EyreSite ->
+  KingSubsite ->
+  W.Application
+app env who liv inform cancel site sub req respond = do
+  met <- maybe (error "bad method") pure (cookMeth req)
```

Comment:
This doesn't have to be monadic

```suggestion
  let met <- maybe (error "bad method") id (cookMeth req)
```

and in fact you can even move it into the where block I suggest below, if you feel like it (or not)